File: MyDailyBeat/MyDailyBeat/EVCFirstTimeSetupViewController.m
//
//  EVCFirstTimeSetupViewController.m
//  MyDailyBeat
//
//  Created by Virinchi Balabhadrapatruni on 7/12/16.
//  Copyright © 2016 eVerveCorp. All rights reserved.
//

#import "EVCFirstTimeSetupViewController.h"
#import "EVCFirstTimePreferencesViewController.h"

@interface EVCFirstTimeSetupViewController ()

@end

@implementation EVCFirstTimeSetupViewController

@synthesize api;

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    api = [RestAPI getInstance];
    [self.message setText:@"Welcome to MyDailyBeat! Before you begin, please set the following preferences."];
    
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)next:(id)sender {
    VervePreferences *prefs = [[VervePreferences alloc] init];
    dispatch_queue_t queue = dispatch_queue_create("dispatch_queue_t_dialog", NULL);
    dispatch_async(queue, ^{
        dispatch_async(dispatch_get_main_queue(), ^{
            [self.view makeToastActivity];
        });
        prefs.userPreferences = [api getUserPreferencesForUser:[api getCurrentUser]];
        prefs.matchingPreferences = [api getMatchingPreferencesForUser:[api getCurrentUser]];
        prefs.hobbiesPreferences = [api getHobbiesPreferencesForUserWithScreenName:[api getCurrentUser].screenName];
        dispatch_async(dispatch_get_main_queue(), ^{
            [self.view hideToastActivity];
            EVCFirstTimePreferencesViewController *prefsView = [[EVCFirstTimePreferencesViewController alloc] initWithNibName:@"EVCFirstTimePreferencesViewController" bundle:nil];
            [self.navigationController pushViewController:prefsView animated:YES];
        });
    });
    
}

@end

File: MDB_Swift/MyDailyBeat/MyDailyBeat/src/EVCFirstTimeSetupViewController.swift

//
//  EVCFirstTimeSetupViewController.swift
//  MyDailyBeat
//
//  Created by Virinchi Balabhadrapatruni on 7/12/16.
//  Copyright © 2016 eVerveCorp. All rights reserved.
//
import UIKit
import Toast_Swift
import API
class EVCFirstTimeSetupViewController: UIViewController {
    var api: RestAPI!
    @IBOutlet var message: UILabel!
    @IBOutlet var nextButton: UIButton!

    @IBAction func next(_ sender: Any) {
        let prefs = VervePreferences()
        
        DispatchQueue.global().async(execute: {() -> Void in
            DispatchQueue.main.async(execute: {() -> Void in
                self.view.makeToastActivity(ToastPosition.center)
            })
            prefs.userPreferences = self.api.getUserPreferences()
            prefs.matchingPreferences = self.api.getMatchingPreferences()
            prefs.hobbiesPreferences = self.api.getHobbiesPreferencesForUser()
            DispatchQueue.main.async(execute: {() -> Void in
                self.view.hideToastActivity()
                self.performSegue(withIdentifier: "PrefsSegue", sender: nil)
            })
        })
    }


    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        api = RestAPI.getInstance()
        self.message.text = "Welcome to MyDailyBeat! Before you begin, please set the following preferences."
    }
    
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        if let dest = segue.destination as? EVCPreferencesViewController {
            dest.onSuccess = { (success, vc) in
                vc.dismiss(animated: true, completion: nil)
            }
        }
    }

    
}
